lib/containers/runtime.pm

```diff
@@ -131,18 +222,178 @@ sub remove_image {
     $self->_rt_assert_script_run("rmi -f $image_name");
 }
 
+=head2 remove_container
+
+Remove a container from the pool.
+
+=cut
+sub remove_container {
+    my ($self, $container_name) = @_;
+    $self->_rt_assert_script_run("rm -f $container_name");
+}
+
+=head2 check_image_in_host_registry
+
+Returns true if host contains C<img> or false.
+
+=cut
+sub check_image_in_host_registry {
+    my ($self, $img) = @_;
+    my @lregistry = $self->enum_images();
+    $img =~ @lregistry;
+}
+
+=head2 setup_registry
+
+Updates the registry files for the running container runtime to allow access to
+insecure registries.
+
+Implementation is subject to the the subclass otherwise call to this subroutine dies.
+
+=cut
+sub setup_registry {
+    my ($self) = shift;
+    die "Unsupported runtime - " . $self->runtime;
```

Comment:
Why die? Just  use `return`. If a runtime doesn't need to configure anything, you should let this function succeed.

Reply:
there was a `die "You must define the runtime!" unless $runtime;` but checking now this likely doesnt make sense here. will remove

Reply:
*this change is ready for review*